### PR TITLE
btcstaking: query finality provider

### PR DESCRIPTION
**Pull request by SebastianElvis**
This PR adds a query to finality provider with the given BTC PK.

Context: https://babylon-chain.slack.com/archives/C06CJ0J6UJ3/p1707884423920019?thread_ts=1707882635.267279&cid=C06CJ0J6UJ3

### CODE REVIEW

File: x/btcstaking/client/cli/query.go [hunk 1]
```diff
@@ -33,6 +34,33 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
...
+		Use:   "finality-provider [fp_btc_pk_hex]",
+		Short: "retrieve a finality provider",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			queryClient := types.NewQueryClient(clientCtx)
+			res, err := queryClient.FinalityProvider(
+				cmd.Context(),
+				&types.QueryFinalityProviderRequest{
+					FpBtcPkHex: args[0],
```
Comment (gitferry) [inline x/btcstaking/client/cli/query.go:48]: Do we need to verify the pk first?

**Review by gitferry — APPROVED** [review]
Lgtm!

File: x/btcstaking/client/cli/query.go (hunk 1 above)
Comment (SebastianElvis) [reply x/btcstaking/client/cli/query.go:48]: Hmm I would say it's not necessary. The node side will return error if the pk is malformed anyway. It's a query so does not have implication on gas. Wdyt?

Comment (gitferry) [reply x/btcstaking/client/cli/query.go:48]: I see. Since we are passing the PK as string type, we need to check the type anyway. All good then

**Pull request merged**